widgets/common/src/hooks/useRefreshEvent.js

```diff
@@ -0,0 +1,14 @@
+import { useEffect } from 'react';
+
+function useRefreshEvent(toolbox, event, handler = toolbox.refresh) {
```

Comment:
I'm in two minds about this name. From one side, yeah, the default value now makes sense, due to the name of the function.

From the other side, there is nothing in this implementation that prevents consumers from using it with any event, and doing any action. Having a specific name here (related to refreshing) in my mind is more confusing if the hook is used in other contexts (in which it functionally works well).

Having thought about it, I would suggest keeping the old name, so as not to limit the usage contexts, and removing the default value, as mentioned before :slightly_smiling_face: 

```suggestion
function useEventListener(toolbox, event, handler) {
```

Reply:
Alternatively, we could do something to make sure this hook is not _hackably_ used in some other context, to listen to a non-refresh event

Reply:
I'd say it's hard to agree. Here are my points:

1. Function name should reflect its implementation as close as possible, that's obvious. However, it should be the implementation that should drive the name, not the other way around.

2. After reviewing events usage in widgets I found out that almost all events handled are `[resourceName]:refresh`. There is only one exception which is `topology:selectNode` event. It would still make sense to use this event with hook named `useRefreshEvent` as we need to refresh widget state to reflect application state change. On the other hand, if we used this event with hook named `useEventListener` the `toolbox.refresh` default handler would still make sense for the same reason - we need to refresh widget state to reflect application state change.

Summing up I'm OK with both proposed names, but I can see no good reason for removing the default handler.

Reply:
Hmm... I can see how removing the default handler can be inconvenient. I still would not expect `toolbox.refresh` to be the default handler when seeing:

```ts
useEventListener(toolbox, 'abc:refresh');
```

How about exposing 2 hooks?

1. `useEventListener` without the default value, for general purposes
2. `useRefreshEvent` that would call `toolbox.refresh` automatically inside and not expose any `handler`

That could solve our problems with an unexpected default value, and with repeating `toolbox.refresh` multiple times

Reply:
Makes perfect sense now, implemented.

Reply:
Great :tada: I like it!